src/vs/workbench/contrib/terminal/electron-browser/terminalInstance.ts

```diff
@@ -1063,12 +1093,24 @@ export class TerminalInstance implements ITerminalInstance {
 
 		// Set the new shell launch config
 		this._shellLaunchConfig = shell; // Must be done before calling _createProcess()
-		// Initialize new process
-		this._createProcess();
+
+		// Launch the process unless this is only a renderer.
+		if (!this._shellLaunchConfig.isRendererOnly) {
+			this._createProcess();
+		}
+
 		if (oldTitle !== this._title) {
 			this.setTitle(this._title, true);
 		}
-		this._processManager!.onProcessData(data => this._onProcessData(data));
+
+		if (this._processManager) {
+			// TODO: Why is this a string-null check failure without the "!"?
```

Comment:
Removing the ! should work fine within the if, are you sure you saved the file? Strict null check errors need the file to be saved as it's a special tsconfig.